src/tasks/run_center_out_reach.py

```diff
@@ -336,22 +341,40 @@ def run():
             "black", window_settings.colors.target, user_input.input_device_name
         )
 
-    with open_connection(data_output), open_connection(data_input):
-        task_window = TaskWindow(window_rect, window_params, menu_text)
-        task_state = TaskState(task_window, state_params)
-        task_runner = TaskRunner(
-            sampling_rate,
-            data_input,
-            data_output,
-            velocity_scaler,
-            with_decoded_cursor,
-            metrics_collector,
-        )
-        task_runner.run(task_state, user_input)
-
-        if not task_window.show_menu_screen and _metrics_enabled(settings):
-            unwrap(metrics_collector).plot_metrics(task_window.target_positions)
-
+    interrupted = False
+    task_window = None
+    try:
+        with open_connection(data_output), open_connection(data_input):
+            task_window = TaskWindow(window_rect, window_params, menu_text)
+            task_state = TaskState(task_window, state_params)
+            task_runner = TaskRunner(
+                sampling_rate,
+                data_input,
+                data_output,
+                velocity_scaler,
+                with_decoded_cursor,
+                metrics_collector,
+            )
+            logger.info("Running task")
+            task_runner.run(task_state, user_input)
+    except KeyboardInterrupt:
+        logger.info("CTRL+C received. Exiting...")
+        interrupted = True
+
+    # This is used as a signal to a parent process that the main task has finished
+    if args.pipe is not None:
+        with open(args.pipe, "w") as pipe:
+            pipe.writelines(["main_task_finished"])
```

Comment:
```suggestion
            pipe.write("main_task_finished")
```

Reply:
I just realized that the writelines do not add a newline 😨 
I will use your suggestion adding the `\n` to the end. Thanks!